.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,17 @@
+name: Set up Python
+description: Installs Python3 and dependencies defined in the Pipfile
+runs:
+  using: composite
+  steps:
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.9'
```

Comment:
Note that now in some scenarios and places where this DRY action will now be used, we were just saying 3.x and now we're saying 3.9. I don't think that should be an issue but wanted to call that out.

Reply:
This was one of the reasons for doing this - specifically because we had different call outs for version across our actions. Now it's locked in. Also if we wanted to do testing; we can introduce a version `matrix` and run things with multiple versions if needed.